### ISSUE TITLE
Admonish doesn't work with `mdbook test`

### ISSUE THREAD

**Issue by stevecheckoway** [tommilligan/mdbook-admonish#108]
If I use an admonish example (or other box) containing a code example, `mdbook test` will not test that example.
Here's an example of an example that doesn't get tested.

`````````
``````admonish example
Here's an example
```rust
let x = 10;
x = 20;
```
``````
`````````

If I take it out of an `admonish example` fenced code block, the code is tested (and fails as expected).

Is there a way to make this work?

**Comment by tommilligan** [comment]
Hmm, interesting limitation. My guess is that `mdbook test` skips any code blocks that are not considered rust source code (i.e. have the info string `rust`, or have an empty info string). My intuition is that you'd have to change the behaviour of `mdbook` to look inside code blocks for other code blocks to test (which I think is bad, and probably won't be accepted upstream).

As `mdbook-admonish` respects `import` statements, I think you could work around this by putting your code snippets in another file, importing it inside `admonish`, and then importing it _again_ not inside an admonish. The plain import should get picked up and tested.

Note that you may need to first [tweak your book.toml settings](https://github.com/tommilligan/mdbook-admonish#process-included-files) to make includes expand in the right order.

So something like:

`````markdown
# My nice book page

Code on this page is not tested

````admonish
```rust
{{#include some_rust_code.rs}}
```
````
`````

`````markdown
# My secret test page

Code on this page is tested

```rust
{{#include some_rust_code.rs}}
```
`````

```rust
// some_rust_code.rs
let x = 10;
x = 20;
```

Something like this works for me locally, though I agree it's a bit of a pain.

**Comment by stevecheckoway** [comment]
I just had a thought about this. It looks like `mdbook test` will run preprocessors, but just those that support the "test" renderer, [I think anyway](https://github.com/rust-lang/mdBook/blob/581e5025a2342353a69a23152b255fd7819f8f10/src/book/mod.rs#L273).

I have to run and haven't had a chance to try this, but I thought I'd mention it since it might be as simple as having mdbook-admonish support the test renderer. I'll try to test it later today.

**Comment by tommilligan** [comment]
Ah interesting. So I think you're right, I'd forgotten this system where we know the type of rendering that's being done.

If we're in `test` mode, we could simply strip out all admonish-specific syntax, leaving the bare internals behind. Then, if they're code blocks, they should be picked up afterwards.

I'll have a look at that now

**Comment by tommilligan** [comment]
This appears to support your use case - take a look and let me know any thoughts: https://github.com/tommilligan/mdbook-admonish/pull/109

**Comment by tommilligan** [comment]
Released in [v1.10.0](https://github.com/tommilligan/mdbook-admonish/releases/tag/v1.10.0)
